DAS_Sistemas Develop Push: Practica-2

JesusAGC opened the pull request.

JesusAGC committed:
Commit de practica 2

AnhellO commented on Ago-Dic-2020/gonzalez-cardenas-jesus-antonio/Practica 2/Practica2.py:
```diff
@@ -0,0 +1,95 @@
+    Cuenta = 0.000
+    cobro = 0.000
+    def __init__ (self, name, dateRelease, passengers, numOfWheels, terrain, typeVehicle,):
+        Vehiculo.__init__(self, name, dateRelease, passengers, numOfWheels, terrain, typeVehicle)
```
Te recomiendo utilizar la función [`super()`](https://realpython.com/python-super/) 😉

AnhellO commented on Ago-Dic-2020/gonzalez-cardenas-jesus-antonio/Practica 2/Practica2.py:
```diff
@@ -0,0 +1,95 @@
+class Camion(Vehiculo):
+    NumActualDePasajeros = 0
+    Cuenta = 0.000
+    cobro = 0.000
```
No es necesario definir estas variables de clase, ya que estas son diferentes de los atributos o variables de instancia: https://www.digitalocean.com/community/tutorials/understanding-class-and-instance-variables-in-python-3

AnhellO commented on Ago-Dic-2020/gonzalez-cardenas-jesus-antonio/Practica 2/Practica2.py:
```diff
@@ -0,0 +1,95 @@
+class Vehiculo(object):
+    def __init__ (self, name, dateRelease, passengers, numOfWheels, terrain, typeVehicle):
```
Hay un espacio extra entre `__init__` y el paréntesis que abre `(`

AnhellO replied:
Otro comentario es que las variables tienen que ser estilo `snake_case` de acuerdo al estándar oficial del lenguaje: https://www.python.org/dev/peps/pep-0008/ (lo mismo aplica para las funciones). Esto aplica para todas las funciones y variables de la práctica que no cumplan con esto

AnhellO commented on Ago-Dic-2020/gonzalez-cardenas-jesus-antonio/Practica 2/Practica2.py:
```diff
@@ -0,0 +1,95 @@
+        self.typeVehicle = typeVehicle
+
+    def specs(self):
+        print(self.name+' salió en la fecha '+ self.dateRelease +', tiene el número de llantas: ' +str(self.numOfWheels) +'. Y es para el terreno tipo: ' +self.terrain)
```
Te recomiendo utilizar el [`f string`](https://realpython.com/python-f-strings/), así evitarás múltiples concatenaciones y con ello se mejorará el performance 👍

JesusAGC committed:
Commit modificaciones de practica 2

JesusAGC committed:
Commit modificaciones de practica 2 - 2

AnhellO commented on Ago-Dic-2020/gonzalez-cardenas-jesus-antonio/Practica 2/classes.dot:
```diff
@@ -0,0 +1,9 @@
+digraph "classes" {
```
Hay que actualizar el diagrama debido a que los nombres de los atributos de clase cambiaron

AnhellO reviewed (changes requested):
Un comentario más

JesusAGC committed:
Commit de practica 2 - 3
Se cambio el nombre de los métodos para que cumplan con el estilo snake_case y se corrigio el .dot que para que tenga los nuevos nombres de variables y métodos

JesusAGC committed:
Commit de practica 2 - 3.1
Añado archivos que faltaron en el commit anterior
